lib/context/plan-context.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const INTERNAL = Symbol("internal");
+
+class PlanContext {
+    constructor(plan) {
+        this[INTERNAL] = {};
+        this[INTERNAL].plan = plan;
+    }
+
+    set state(value) {
+        this[INTERNAL].state = value;
+    }
+    
+    get state() {
+        return this[INTERNAL].state;
+    }
+
+    set current(value) {
+        this[INTERNAL].current = value;
+    }
+    
+    get current() {
+        return this[INTERNAL].current;
+    }
+
+    set originalInput(value) {
```

Comment:
nice to have: we should make it so you cannot set `originalInput` after its been set (so maybe no setter?)